src/download.rs

```diff
@@ -123,7 +124,11 @@ pub fn get_latest_version(name: &str) -> Result<Version> {
             &OfficialToolchainDescription::from_str("latest")?,
             tmp_dir.into_path(),
         ) {
-            Ok(channel.pkg[name].version.clone())
+            if let Some(p) = channel.pkg.get(name) {
+                Ok(p.version.clone())
+            } else {
+                bail!("'{name}' is not a valid, downloadable package in the 'latest' channel.")
+            }
```

Comment:
This could also be written as something like:
```rust
channel.pkg.get(name)
    .ok_or_else(|| anyhow!("'{name}' is not a valid, downloadable package in the 'latest' channel."))?
```